DeusEx/Classes/DXRPasswords.uc

```diff
@@ -294,6 +352,113 @@ function LogAll()
     }
 }
 
+function ProcessText(DeusExTextParser parser, out int hasPass[64])
+{
+    local string text;
+    local int i;
+    local byte tag;
+
+    while(parser.ProcessText()) {
+        tag = parser.GetTag();
+        if( tag != 0 ) continue;
+
+        text = Caps(parser.GetText());
+        if( Len(text) == 0 ) continue;
+
+        for(i=0; i<passEnd; i++) {
+            if( Len(oldpasswords[i]) == 0 ) continue;
+            if( WordInStr( text, Caps(oldpasswords[i]), Len(oldpasswords[i]), true ) != -1 ) {
+                hasPass[i] = 1;
+                //l("hasPass["$i$"] = 1;");
+            }
+        }
+    }
+}
+
+function bool CheckComputerPosition(InformationDevices id, Computers c, vector newpos, int hasPass[64])
+{
+    local int a, i, num;
+
+    if( PositionIsSafe(id.Location, c, newpos) ) return True;// don't even need to check the passwords
+
+    for (i=0; i<ArrayCount(c.userList); i++)
+    {
+        if (c.userList[i].password == "")
+            continue;
+        num++;
+    }
+    if ( num == 0 ) return True;
+    if ( id.textTag == '' ) return False;
+
+    for (i=0; i<ArrayCount(c.userList); i++)
+    {
+        if (c.userList[i].password == "")
+            continue;
+        
+        for (a=0; a<passEnd; a++) {
+            if( hasPass[a]==1 && c.userList[i].password == newpasswords[a] ) {
```

Comment:
should I use Caps or ~= ?

Reply:
don't need it, my own code sets the computer's password to be exactly equal to the newpasswords